src/api-next/api-next.ts

```diff
@@ -1697,4 +1700,94 @@ export class IBApiNext {
         .pipe(map((v: { all: HistogramEntry[] }) => v.all))
     );
   }
+
+  /**
+   * Feeds in currently open orders.
+   *
+   * @param listener
+   * orderId: The order's unique id.
+   *
+   * contract: The order's [[Contract]]
+   *
+   * order: The currently active [[Order]]
+   *
+   * orderState: The order's [[OrderState]]
+   *
+   * @see [[placeOrder]], [[reqAllOpenOrders]], [[reqAutoOpenOrders]]
+   */
+  private readonly onOpenOrder = (
+    subscriptions: Map<number, IBApiNextSubscription<OpenOrder[]>>,
+    orderId: number,
+    contract: Contract,
+    order: Order,
+    orderState: OrderState
+  ): void => {
+    subscriptions.forEach((sub) => {
+      const allOrders = sub.lastAllValue ?? [];
+      allOrders.push({ orderId, contract, order, orderState });
+      sub.next({
+        all: allOrders,
+      });
+    });
+  };
+
+  /**
+   *  Ends the subscrition once all openOrders are recieved
+   *  @param subscriptions
+   */
+  private readonly onOpenOrderEnd = (
+    subscriptions: Map<number, IBApiNextSubscription<OpenOrder[]>>
+  ): void => {
+    console.log("Recieved end of Orders");
+    subscriptions.forEach((sub) => {
+      sub.complete();
+    });
+  };
+
+  /**
+   * Response to API bind order control message.
+   *
+   * @param orderId: permId
+   * @param apiClientId: API client id.
+   * @param apiOrderId: API order id.
+   *
+   * @see [[reqOpenOrders]]
+   */
+  private readonly onOrderBound = (
+    // TODO finish implementation
+    subscription: Map<number, IBApiNextSubscription<OpenOrder[]>>,
+    orderId: number,
+    apiClientId: number,
+    apiOrderId: number
+  ): void => {
+    // not sure what it's used for
+    console.log(
+      `missing OrderBound implementation - ${orderId}, ${apiClientId}, ${apiOrderId}`
+    );
+  };
+
+  /**
+   * Requests all current open orders in associated accounts at the current moment.
+   * The existing orders will be received via the openOrder and orderStatus events.
+   *
+   * Open orders are returned once; this function does not initiate a subscription.
+   * Author: Tsopic - proud father of his first rxjs function
```

Comment:
Not correct anymore, would change to simply "Requests all current open orders in associated accounts at the current moment."